plugins/serializers/msgpack/msgpack.go

```diff
@@ -0,0 +1,51 @@
+package msgpack
+
+import (
+	"github.com/influxdata/telegraf"
+)
+
+// Serializer encodes metrics in MessagePack format
+type Serializer struct {
+}
+
+// NewSerializer creates a msgpack.Serializer
+func NewSerializer() *Serializer {
+	return &Serializer{}
+}
+
+func marshalMetric(buf []byte, metric telegraf.Metric) ([]byte, error) {
+	return (&Metric{
+		Name:   metric.Name(),
+		Time:   MessagePackTime{time: metric.Time()},
+		Tags:   metric.Tags(),
+		Fields: metric.Fields(),
+	}).MarshalMsg(buf)
+}
+
+// Serialize implements serializers.Serializer.Serialize
+// github.com/influxdata/telegraf/plugins/serializers/Serializer
+func (s *Serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	buf, err := marshalMetric(nil, metric)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return buf, nil
```

Comment:
You might even want to do
```suggestion
	return marshalMetric(nil, metric)
```
or even directly move the `marshalMetric()` function in there.

Reply:
Ack.